### PR TITLE
Fix up test in PR

### PR DESCRIPTION
**Pull request by patocallaghan**
@marvinmarnold I had a look and I fixed up the test in your [PR](https://github.com/gothinkster/ember-realworld/pull/112).

It appears the test wasn't working correctly because Mirage wasn't set up correctly to handle the `/users` POST endpoint for registration, and the error was being swallowed.

Feel free to just merge this into your branch once you're happy with it.

### CODE REVIEW

File: app/services/session.js [hunk 1]
```diff
@@ -45,8 +45,8 @@ export default class SessionService extends Service {
     try {
       await user.save();
       this.setToken(user.token);
-    } catch {
-      // Registration returned errors
+    } catch (e) {
+      console.error(e);
```
Comment (patocallaghan) [inline app/services/session.js:49]: The Mirage error happening at this point was just being swallowed before

File: mirage/config.js [hunk 2]
```diff
@@ -100,7 +100,10 @@ export default function() {
   /**
    * User registration
    */
-  // this.post('/users', (schema, request) => {});
+  this.post('/users', (schema, request) => {
```
Comment (patocallaghan) [inline mirage/config.js:103]: We needed to set up this endpoint for user registrations

File: tests/acceptance/register-test.js [hunk 3]
```diff
@@ -11,23 +11,24 @@ module('Acceptance | register', function(hooks) {
   setupLoggedOutUser(hooks);
 
   test('successful registration', async function(assert) {
-    const user = {
+    const user = this.server.create('user', {
```
Comment (patocallaghan) [inline tests/acceptance/register-test.js:14]: Use Mirage to prepopulate the db and add a `username` as that's what's being rendered in the nav

**Comment by marvinmarnold** [comment]
This is awesome, thanks! @patocallaghan

**Pull request merged**